AppCenter/AppCenter/Internals/HttpClient/MSHttpClientPrivate.h

```diff
@@ -39,15 +36,23 @@
  */
 @property(nonatomic, getter=isEnabled) BOOL enabled;
 
-/**
- * A boolean value set to YES if payload compression is enabled or NO otherwise.
- */
-@property(nonatomic, getter=isCompressionEnabled) BOOL compressionEnabled;
-
 /**
  * Configuration object for the NSURLSession. Need to store this because the session will need to be re-created after re-enabling the
  * client.
  */
 @property(nonatomic) NSURLSessionConfiguration *sessionConfiguration;
 
+/**
+ * Hash table containing all the delegates as weak references.
+ */
+@property NSHashTable<id<MSHttpClientDelegate>> *delegates;
+
+/**
+ * Disables the client, deletes data, and cancels any calls.
+ *
+ * @param isEnabled Whether to enable or disable the client.
+ * @param deleteData Whether to delete data on disabled.
+ */
+- (void)setEnabled:(BOOL)isEnabled andDeleteDataOnDisabled:(BOOL)deleteData;
```

Comment:
This should be remove?

Reply:
@jaelim-ms This is used by 2 tests in MSHttpClientTests. `testDisablingCancelsCalls` and `testDisableThenEnable`. Can it actually be removed with those tests?